dotcom-rendering/src/web/browser/islands/doHydration.ts

```diff
@@ -15,19 +14,19 @@ import { initPerf } from '../initPerf';
  * @param data The deserialised props we want to use for hydration
  * @param element The location on the DOM where the component to hydrate exists
  */
-export const doHydration = (
+export const doHydration = async (
 	name: string,
-	data: Attributes | null,
+	data: { [key: string]: unknown } | null,
 	element: HTMLElement,
-): void => {
+): Promise<void> => {
 	// If this function has already been run for an element then don't try to
 	// run it a second time
 	const alreadyHydrated = element.dataset.guReady;
 	if (alreadyHydrated) return;
 
 	const { start, end } = initPerf(`hydrate-${name}`);
 	start();
-	import(
+	return import(
```

Comment:
I think this is refactor that is orthogonal to this change. Could it be split in its own PR?

Making this function async is not really giving us anything as we are not awaiting it anywhere.

Reply:
`await import ...` is probably a nicer read if that's an option

Reply:
Would definitely support this refactor :)